pkg/scalers/kafka_scaler.go

```diff
@@ -398,6 +479,22 @@ func getKafkaClients(metadata kafkaMetadata) (sarama.Client, sarama.ClusterAdmin
 		config.Net.SASL.TokenProvider = OAuthBearerTokenProvider(metadata.username, metadata.password, metadata.oauthTokenEndpointURI, metadata.scopes, metadata.oauthExtensions)
 	}
 
+	if metadata.saslType == KafkaSASLTypeGSSAPI {
+		config.Net.SASL.Enable = true
+		config.Net.SASL.Mechanism = sarama.SASLTypeGSSAPI
+		config.Net.SASL.GSSAPI.ServiceName = "kafka"
```

Comment:
Should this be a parameter?

Reply:
Don't believe this needs to be exposed to users - no value in it.  Scaler process will only need a single Kafka service.

Reply:
Needing to enable sasl can be mentioned in the docs. Giving the end user more control rather than some gotcha of it being automatically enabled.

Reply:
Nothing automatic is happening.  User specifies option - `sasl: gssapi`.  This way they declare intent to turn on sasl with kerberos auth.